### ISSUE TITLE
[epic] Deferred normalization ASTs 

### ISSUE THREAD

**Issue by rbalicki2** [isographlabs/isograph#43]
* Depends on #33

## Background: entrypoint artifacts and normalization ASTs

* For each `` iso(`entrypoint Type.Field`) `` literal that the compiler encounters, it will generate a `Type/Field/entrypoint.ts` file.
* This file currently (i.e. pre #33) contains a query text and a normalization AST.
* The query text is used when making the query
* The normalization AST is (currently) only used to write the network response into the key-value store and for garbage collection (which is also only done after the network response is received).
     * They will (after #42) be used to determine whether a network request needs to be made.
* Because the normalization AST is only used after a network response is received, we can fetch the normalization AST later (e.g. when making the network request).

## This epic

* This epic is to load normalization ASTs when the network request is created, via https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import or something.
* This should be controllable with a directive on the entrypoint
    * So, the iso literal parser should be modified to include support for parsing directives
* Whether the normalization AST is hard-loaded should be reflected in the type of the entrypoint, since the entrypoint **cannot** be used with an `if-necessary` fetch policy if the normalization AST is not present.
    * One should be able to eventually call `entrypoint.loadNormalizationAst().then(entrypointWithNormalizationAst => fetchIfNecessary(...))` or the like

**Comment by rbalicki2** [comment]
I think the remaining tasks are:
- [x] move normalization ASTs into their own files
- [x] add support for a @lazyLoadAst directive (or something) to entrypoint literals
- [ ] replacing the hard require of the normalization AST file with `() => import('./normalizationAst')` when that directive is encountered

**Comment by XiNiHa** [comment]
I'd like to work on this one!
